src/Entity/AuditRecord.php

```diff
@@ -57,6 +73,40 @@ private function __construct(
         $this->datetime = new \DateTimeImmutable();
     }
 
+    public function getIp(): ?string
+    {
+        return $this->ip;
+    }
+
+    public function setIp(?string $ip): void
+    {
+        if ($this->locked) {
+            throw new \LogicException('Cannot modify IP after entity has been persisted');
+        }
+
+        $this->ip = $ip;
+    }
+
+    public function getUserAgent(): ?string
+    {
+        return $this->userAgent;
+    }
+
+    public function setUserAgent(?string $userAgent): void
+    {
+        if ($this->locked) {
+            throw new \LogicException('Cannot modify User-Agent after entity has been persisted');
+        }
+        $this->userAgent = $userAgent;
+    }
+
+    #[ORM\PostPersist]
+    #[ORM\PostLoad]
+    public function lock(): void
+    {
+        $this->locked = true;
+    }
```

Comment:
Could also move this into the `AuditRecordListener` so we don't start mixing different approaches, but we need a `lock()` method that we can call from within that listener.